src/util/range.h

```diff
@@ -341,6 +341,32 @@ struct ranget final
     return begin_value == end_value;
   }
 
+  /// Drop the first \p count elements.
+  /// If the range has fewer elements, drops everything.
+  /// Similar to \ref skip but modifies the range in-place.
+  /// \return true if the range still contains element after the drop
+  bool drop(std::size_t count)
+  {
+    for(std::size_t i = 0; i < count; ++i)
```

Comment:
You don't actually need `i`, do you? `for( ; count > 0; --count) ...` would do the trick.